javatests/google/registry/rde/imports/RdeDomainImportActionTest.java

```diff
@@ -135,6 +148,151 @@ public void test_mapreducePublishesToDns() throws Exception {
     assertDnsTasksEnqueued("example1.test");
   }
 
+  /**
+   * Verifies the state of an imported pending transfer before and after implicit server approval
+   */
+  @Test
+  public void testMapreducePendingTransferServerApproval() throws Exception {
+    pushToGcs(DEPOSIT_1_DOMAIN_PENDING_TRANSFER);
+    runMapreduce();
+    List<DomainResource> domains = ofy().load().type(DomainResource.class).list();
+    assertThat(domains).hasSize(1);
+    checkDomain(domains.get(0));
+    // implicit server approval happens at 2015-01-08T22:00:00.0Z
+    DateTime serverApprovalTime = DateTime.parse("2015-01-08T22:00:00.0Z");
+    // Domain should be assigned to RegistrarX before server approval
+    DomainResource beforeApproval =
+        domains.get(0).cloneProjectedAtTime(serverApprovalTime.minus(Seconds.ONE));
+    assertThat(beforeApproval.getCurrentSponsorClientId()).isEqualTo("RegistrarX");
+    assertThat(loadAutorenewBillingEventForDomain(beforeApproval).getClientId())
+        .isEqualTo("RegistrarX");
+    assertThat(loadAutorenewPollMessageForDomain(beforeApproval).getClientId())
+        .isEqualTo("RegistrarX");
+    // Current expiration is 2015-04-03T22:00:00.0Z
+    assertThat(beforeApproval.getRegistrationExpirationTime())
+        .isEqualTo(DateTime.parse("2015-04-03T22:00:00.0Z"));
+    // Domain is not yet in transfer grace period
+    assertThat(beforeApproval.getGracePeriodStatuses()).doesNotContain(GracePeriodStatus.TRANSFER);
+    // Check autorenew events - recurrence end is set to transfer approval time,
+    // and client id is set to losing registrar. This is important in case the
+    // transfer is cancelled or rejected.
+    // Event time is set to domain expiration date.
+    checkAutorenewBillingEvent(beforeApproval,
+        "RegistrarX",
+        DateTime.parse("2015-04-03T22:00:00.0Z"),
+        DateTime.parse("2015-01-08T22:00:00.0Z"));
+    checkAutorenewPollMessage(beforeApproval,
+        "RegistrarX",
+        DateTime.parse("2015-04-03T22:00:00.0Z"),
+        DateTime.parse("2015-01-08T22:00:00.0Z"));
+
+    // Domain should be assigned to RegistrarY after server approval
+    DomainResource afterApproval =
+        domains.get(0).cloneProjectedAtTime(serverApprovalTime.plus(Seconds.ONE));
```

Comment:
No need to add the extra second; the transfer is effective precisely at the server approval time, inclusive.  See e.g. the test here:
https://github.com/google/nomulus/blob/ec55aa5361f71d86682817252054cbaf2caf9c7c/javatests/google/registry/flows/domain/DomainTransferRequestFlowTest.java#L242

Better to be precise to avoid giving the impression that the boundary is fuzzy.